docs/design/safety/README.md

```diff
@@ -0,0 +1,220 @@
+to our definition:
+
+-   Carbon might choose to _not_ prevent data race bugs that are not
+    _themselves_ also temporal safety bugs, even though the data race may lead
+    to corruption and cause the program to later violate various other forms of
+    memory safety. So far, evidence has not shown this to be as significant and
+    prevalent source of _vulnerabilities_ as other forms of memory safety bugs.
+-   However, Carbon _must_ detect and prevent cases where a lack of
+    synchronization directly allows temporal safety bugs, such as use after
+    free.
+
+However, preventing these data race bugs remains _highly valuable_ for
+correctness, debugging, and achieving [fearless concurrency]. But it is one aspect
+where Carbon can in theory afford a compromise based on the current security information.
+
+[fearless concurrency]: https://doc.rust-lang.org/book/ch16-00-concurrency.html
+
+## Safe library ecosystem
+
+Carbon will need access to a memory safe library ecosystem. The industry is
+currently investing a massive amount of effort to build out a sufficient
+ecosystem of such software using Rust, and it is critical that Carbon does not
+impede, slow down, or require duplicating that ecosystem.
+
+**Carbon's strategy for safe and generally reusable libraries is to leverage
+Rust libraries through interop.** This is a major motivating reason for seamless
+and safe Rust interop. The Carbon project will work to avoid creating
+duplication between the growing Rust library ecosystem and any future Carbon
+library ecosystem. Carbon's ecosystem will be focused instead on libraries and
+functionality that would either be missing or only available in C++.
+
+## Build modes
+
+Where Carbon's safety mode governs the language rules applied to unsafe code,
+Carbon's build modes will change the _behavior_ of unsafe code.
+
+There are two primary build modes:
+
+-   **Release**: the primary build mode for programs in production, focuses on
+    giving the best possible performance with a practical baseline of safety.
+-   **Debug**: the primary build mode during development, focuses on
+    high-quality developer experience.
+
+The release build will include a baseline of hardening necessary to uphold the
+run-time enforcement components of our
+[memory safety model](#memory-safety-model) above. This means, for example, that
+bounds checking is enabled in the release build. There is [evidence] that the
+cost of these hardening steps is low. Following the specific guidance of our top
+priority for [performance _control_], Carbon will provide ways to write unsafe code
+that disables the run-time enforcement, enabling the control of any overhead incurred.
+
+[evidence]: https://chandlerc.blog/posts/2024/11/story-time-bounds-checking/
+[performance control]: /docs/project/goals.md#performance-critical-software
+
+The debug build will work to cause bugs and any detectable undefined behavior to
```

Comment:
```suggestion
The debug build will work to cause bugs in unsafe code and any detectable undefined behavior to
```

Reply:
I think it would be nice to say that in the text then.

Reply:
Done.